ratpack-pac4j/src/main/java/ratpack/pac4j/internal/Pac4jAuthenticationHandler.java

```diff
@@ -76,4 +78,9 @@ private void initiateAuthentication(final Context context) {
       }
     }).then(ignored -> webContext.sendResponse());
   }
+
+  private boolean isAjaxRequest(Request request) {
+    Headers headers = request.getHeaders();
+    return "XMLHttpRequest".equals(headers.get(HttpHeaderConstants.X_REQUESTED_WITH));
```

Comment:
Might be worthwhile to move `"XMLHttpRequest"` into `HttpHeaderConstants` as well, actually.